src/weather_magic/shaders.cljs

```diff
@@ -29,52 +30,39 @@
   "void main() {
      float lam = lambert(surfaceNormal(vNormal, normalMat),
                          normalize(lightDir));
-     vec4 diffuse = texture2D(base, vUV) +
-                    texture2D(data, mod((vUV - dataPos), 1.0) / dataScale);
-     gl_FragColor = vec4(ambientCol, 1.0) + diffuse * vec4(lightCol, 1.0) * lam;
-   }")
-
-(def blend-fs
-  "void main() {
-     float lam = lambert(surfaceNormal(vNormal, normalMat), normalize(lightDir));
-     vec4 mapDiffuse = texture2D(base, vUV);
-     float temp = texture2D(data, vUV).r * 3.0;
-     gl_FragColor = vec4(ambientCol, 1.0) + mapDiffuse * vec4(lightCol, 1.0) * lam * 0.0001 + vec4(temp, temp, temp, 1.0);
-
-  }")
-
-(def temperature-fs
-  "void main() {
-
-    float temperatureTex1 = texture2D(base, vUV).b;
-    float temperatureTex2 = texture2D(base, vUV).r;
 
-    float temperature = mix(temperatureTex1, temperatureTex2, year/range);
-
-    vec4 outColor;
-
-    float threshold = fov/1000.0;
-    if (fov > 45.0) {
-      threshold = pow((90.0 - fov)/90.0, 3.0)/5.0 - 0.005;
-    }
-
-    float alphaValue = clamp(15.0 / fov, 0.0, 1.0);
-
-    if (mod(temperature, 0.1) < threshold && fov < 50.0 && temperature > 0.15) {
-      if (temperature > 0.5 && temperature < 0.75) {
-        outColor = vec4(0.5, 0.5, 0.5, alphaValue);
-      } else if (temperature > 0.75) {
-        outColor = vec4(0.0, 0.0, 0.0, alphaValue);
-      } else if (temperature < 0.5) {
-        outColor = vec4(1.0, 1.0, 1.0, alphaValue);
-      }
-    } else if(temperature > 0.5) {
-      outColor = vec4(1.0, 1.0 - (2.0 * (temperature - 0.5)), 0, 1.0);
-    } else {
-      outColor = vec4(2.0 * temperature, 2.0 * temperature, 2.0 * (0.5 - temperature), 1.0);
-    }
-    gl_FragColor = outColor;
-  }")
+     float temperature = texture2D(data, mod((vUV - dataPos), 1.0) / dataScale).r;
+     vec4 baseTexture = texture2D(base, vUV);
+
+     vec4 temperatureColor;
+
+     if (mod(temperature, 0.078125) < 0.003 && eye.z < 1.2) {
+       if (temperature > 0.5) {
+         temperatureColor = vec4(0.5, 0.5, 0.5, 1.0);
+       } else if (temperature < 0.5) {
+         temperatureColor = vec4(1.0, 1.0, 1.0, 1.0);
+       }
+     } else if(temperature > 0.5) {
+       temperature = clamp(temperature, 0.5, 0.8);
+       temperatureColor = vec4(1.0,
+                               clamp(-3.333 * temperature + 2.67, 0.0, 1.0),
+                               0.0, 1.0);
+     } else {
+       temperature = clamp(temperature, 0.2, 0.5);
+       temperatureColor = vec4(clamp(6.41 * (temperature-0.34), 0.0, 1.0),
+                               clamp(6.41 * (temperature-0.34), 0.0, 1.0),
+                               clamp((-6.41 * temperature + 3.205), 0.0, 1.0), 
```

Comment:
Verkar bli lite konstiga värden om man sätter in 0.2, tex blir 6.41 * (0.2 - 0.34) negativt.